src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java

```diff
@@ -0,0 +1,115 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                           connection.prepareStatement(selectQuery)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (!resultSet.next()) {
+                return Optional.empty();
+            }
+            return Optional.of(parseFromResultSet(resultSet));
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get driver by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Driver> getAll() {
+        String selectQuery = "SELECT * FROM drivers where deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(selectQuery)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                drivers.add(parseFromResultSet(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all drivers", e);
+        }
+    }
+
+    @Override
+    public Driver update(Driver driver) {
+        String selectQuery = "UPDATE drivers SET name = ?, license_number = ? WHERE id = ?"
+                + " and deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(selectQuery)) {
+            preparedStatement.setString(1, driver.getName());
+            preparedStatement.setString(2, driver.getLicenceNumber());
+            preparedStatement.setLong(3, driver.getId());
+            Driver old = get(driver.getId()).get();
+            preparedStatement.executeUpdate();
+            return old;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update driver " + driver, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String selectQuery = "UPDATE drivers SET deleted = true WHERE id = ?"
+                + " and deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(
+                                selectQuery)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete driver by id " + id, e);
+        }
+    }
+
+    private Driver parseFromResultSet(ResultSet resultSet) throws SQLException {
```

Comment:
you can make this method `package-private static` and use this one in all your daos to not duplicate your code

Reply:
Thanks